packages/template-retail-react-app/app/pages/checkout/partials/shipping-multi-address.jsx

```diff
@@ -0,0 +1,540 @@
+                                                                    postalCode: addr.postalCode
+                                                                }
+                                                            )}
+                                                        </option>
+                                                    ))}
+                                                    <option
+                                                        value="add-new"
+                                                        style={{
+                                                            color: '#3182ce',
+                                                            fontWeight: 'bold'
+                                                        }}
+                                                    >
+                                                        + {formatMessage(addNewAddressLabel)}
+                                                    </option>
+                                                </Select>
+                                            </Box>
+
+                                            {/* Price */}
+                                            <Box
+                                                fontWeight="semibold"
+                                                fontSize="md"
+                                                color="gray.900"
+                                                alignSelf="flex-end"
+                                                mt="auto"
+                                                aria-label={formatMessage({
+                                                    id: 'shipping_multi_address.price.label',
+                                                    defaultMessage: 'Product price'
+                                                })}
+                                            >
+                                                {variant.priceAfterItemDiscount && (
+                                                    <DisplayPrice
+                                                        priceData={getPriceData(variant)}
+                                                        currency={currency}
+                                                        labelForA11y={variant.productName}
+                                                    />
+                                                )}
+                                            </Box>
+                                        </VStack>
+                                    </Flex>
+
+                                    {/* Mobile: Address dropdown and price - only on mobile */}
+                                    <VStack
+                                        align="flex-start"
+                                        w="100%"
+                                        spacing={3}
+                                        display={{base: 'flex', md: 'none'}}
```

Comment:
We have `HideOnDestop` and `HideOnMobile` from `components/responsive` for this purposes